ui/components/component-library/button-icon/button-icon.tsx

```diff
@@ -20,6 +20,7 @@ import {
 } from './button-icon.types';
 
 const buttonIconSizeToIconSize: Record<ButtonIconSize, IconSize> = {
+  [ButtonIconSize.Xs]: IconSize.Xs,
```

Comment:
Hey @ejwessel, this method for adding a size seems correct, but changes to the component-library require a thorough review to ensure alignment with the Figma and mobile component. It also appears that this update isn't currently being used. Should this be removed?

Notice the Figma component does not have an Xs size

![Screenshot 2024-05-23 at 8 32 06 AM](https://github.com/MetaMask/metamask-extension/assets/8112138/43b36261-f451-46a1-a583-1b4061f736b5)